isic-archive-gui/src/Datasets/UploadBatch.vue

```diff
@@ -0,0 +1,198 @@
+          Upload a ZIP file of images.
+        </div>
+        <div class="isic-upload-container">
+          <FileSelect
+            v-model="zipFile"
+            accept=".zip"
+          />
+        </div>
+      </div>
+
+      <div class="isic-form-group">
+        <h3>
+          Licensing
+        </h3>
+        <p>
+          Enter your name to electronically sign the contributor license agreement.
+        </p>
+        <div class="form-group">
+          <label class="control-label">
+            Contributor license agreement
+          </label>
+          <div id="isic-terms-of-use-container" />
+        </div>
+        <div class="form-group">
+          <label
+            class="control-label"
+            for="isic-upload-batch-agreement-signature"
+          >
+            Electronic signature
+          </label>
+          <input
+            id="isic-upload-batch-agreement-signature"
+            v-model.trim="signature"
+            class="form-control input-sm"
+            type="text"
+            placeholder="Enter your name"
+          >
+        </div>
+      </div>
+
+      <form
+        id="isic-upload-batch-form"
+        @submit.prevent="submit"
+      >
+        <button
+          id="isic-upload-batch-submit"
+          class="btn btn-primary btn-md"
+          type="submit"
+          :disabled="!submitReady"
+        >
+          Submit
+        </button>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script>
+import S3 from 'aws-sdk/clients/s3';
+import _ from 'underscore';
+
+import { AccessType } from '@girder/core/constants';
+import { restRequest } from '@girder/core/rest';
+
+import router from '../router';
+import { showAlertDialog } from '../common/utilities';
+
+import DatasetInfoWidget from './DatasetInfoWidget.vue';
+import FileSelect from './FileSelect.vue';
+import SelectDataset from './SelectDataset.vue';
+
+export default {
+  name: 'UploadBatch',
+  components: {
+    DatasetInfoWidget,
+    FileSelect,
+    SelectDataset,
+  },
+  data() {
+    return {
+      dataset: null,
+      zipFile: null,
+      signature: '',
+      uploading: false,
+      AccessType,
+    };
+  },
+  computed: {
+    submitReady() {
+      return this.dataset && this.zipFile && this.signature && !this.uploading;
+    },
+  },
+  methods: {
+    reset() {
+
+    },
+    async submit() {
+      this.uploading = true;
+
+      const initUpload = await restRequest({
+        method: 'POST',
+        url: `dataset/${this.dataset.id}/zip`,
+        data: {
+          signature: this.signature,
+          // TODO: implement a filename parameter on the server
+          filename: this.zipFile.name,
+        },
+      });
+
+      const s3Options = {
+        apiVersion: '2006-03-01',
+        accessKeyId: initUpload.accessKeyId,
+        secretAccessKey: initUpload.secretAccessKey,
+        sessionToken: initUpload.sessionToken,
+      };
+      if (process.env.NODE_ENV !== 'production') {
```

Comment:
@danlamanna To work in a Vagrant + Minio environment, the web client needs to know to use some different options. Would it be reasonable to pass some of this information (or at least provide some sort of `devMode = true`-type) via the upload initialize API?